### ISSUE TITLE
MergeRequest.UpdatedAt is not a *time.Time anymore

### ISSUE THREAD

**Issue by seblegall** [xanzy/go-gitlab#239]
Hello,

I just noticed that `MergeRequest.UpdatedAt` is now type `string` since the upgrade to API v4. Is that a choice or a mistake?

As an example : `Issue.UpdateAt` is still type `time.Time`.

I can work on a PR if this issue is an actual one. What do you think?

**Comment by svanharmelen** [comment]
@seblegall unfortunately this is intentional. Please see issue #209 for more details...

**Comment by seblegall** [comment]
@svanharmelen Wired.

I investigated a little bit. Here is what I found :

* Gitlab API v4 documentation does not mention any crated_at or updated_at field for the `MergeRequest` json object.
* When actually calling the `/merge_requests` endpoint those fields are present
* The date format is ok to be typed as time.Time

Here is an example of what I have : 

```json
[
    {
        "id": 17286,
        "iid": 565,
        "project_id": 341,
        "title": "",
        "description": "",
        "state": "merged",
        "created_at": "2017-11-06T13:17:37.239Z",
        "updated_at": "2017-11-06T13:36:26.063Z",
```

I understand that datetime in general are not consistent in the gitlab API v4. But for this particular example, it seems that the format is ok.

Am I right?

**Comment by svanharmelen** [comment]
I agree it seems like it, but the `MergeRequest` struct is also used in [here](https://github.com/xanzy/go-gitlab/blob/master/events.go#L233) for this API: https://docs.gitlab.com/ce/user/project/integrations/webhooks.html#comment-on-merge-request

The output received from that API call contains a `merge_request` field which is 99% identical to the `MergeRequest` struct, but here the timestamps are in a different format.

So in order to be able to use the struct in both these calls, it seems these fields must be of type `string`. At least until the API uses a consistent timestamp in both API responses.

**Comment by seblegall** [comment]
So sad. :-( I understand your dilemna... Maybe the solution is indeed to duplication the MergeRequest struct. But it's kind a ugly. I'll try to think about a better solution. Maybe there is a way to format the date when Unmarshaling JSON using a custom decoder...

**Comment by svanharmelen** [comment]
Yeah, that should be possible and would be an easy fix for this one. But then the other issue (#227) still remains unsolved. So I think duplicating the struct is (very unfortunately) the better solution in this particular case...

**Comment by svanharmelen** [comment]
Fixed by commit ee48e93a8e3b79669daebe87c77e7b1d40e876f1
